app/dashboard/idena_utils.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+from logging import error
+from pytz import UTC
+from uuid import uuid4
+
+from django.conf import settings
+from django.urls import reverse
+from django.utils import timezone
+
+import requests
+
+from eth_account import Account
+from eth_utils import keccak, decode_hex
+
+from app.services import RedisService
+
+redis = RedisService().redis
+
+def get_idena_url(request, profile):
+    callback_url = request.build_absolute_uri(reverse("profile_by_tab", args=("trust",)))
+    nonce_endpoint = request.build_absolute_uri(reverse("start_session_idena", args=(profile.handle,)))
+    authentication_endpoint = request.build_absolute_uri(reverse("authenticate_idena", args=(profile.handle,)))
+
+    return f'dna://signin/v1?token={profile.idena_token}&'\
+           f'callback_url={callback_url}&' \
+           f'nonce_endpoint={nonce_endpoint}&' \
+           f'authentication_endpoint={authentication_endpoint}&' \
+           f'favicon_url={settings.BASE_URL}static/v2/images/helmet.png&'
+
+def gen_idena_nonce():
+    return f'signin-{uuid4().hex}'
+
+def signature_hash(value):
```

Comment:
Since this function is only used once, and really only does a one-line operation, can we just move the keccak calls into the `signature_address` function so this code is a bit more simple?

Reply:
Yeap that's better!